common/opengl3.cpp

```diff
@@ -628,3 +682,44 @@ std::string fbo::get_status()
 
     return res;
 }
+
+
+void _check_gl_error(const char *file, int line) 
+{
+    GLenum err (glGetError());
+    std::stringstream ss;
+
+    bool has_errors = false;
+    while (err != GL_NO_ERROR) 
+    {
+        std::string error;
+
+        switch (err) 
+        {
+            case GL_INVALID_OPERATION:      error="INVALID_OPERATION";      break;
+            case GL_INVALID_ENUM:           error="INVALID_ENUM";           break;
+            case GL_INVALID_VALUE:          error="INVALID_VALUE";          break;
+            case GL_OUT_OF_MEMORY:          error="OUT_OF_MEMORY";          break;
+            case GL_INVALID_FRAMEBUFFER_OPERATION:  error="INVALID_FRAMEBUFFER_OPERATION";  break;
+        }
+
+        ss << "GL_" << error.c_str() << " - " << file << ":" << line << "\n";
+        err = glGetError();
+        has_errors = true;
```

Comment:
`has_errors` will be inconsistent if in the previous line `err = GL_NO_ERROR`.
Also - all the retrieved values are overwritten by the last one . Is it by design?